packages/core/http/core-http-router-server-internal/src/router.ts

```diff
@@ -204,6 +216,10 @@ export class Router<Context extends RequestHandlerContextBase = RequestHandlerCo
           method,
           path: getRouteFullPath(this.routerPath, route.path),
           options: validOptions(method, route),
+          // For the versioned route security is validated in the versioned router
+          security: internalOptions.isVersioned
+            ? route.security
+            : validRouteSecurity(route.security as DeepPartial<RouteSecurity>),
```

Comment:
Not sure I understand the benefit in handling this validation in the versioned router... Perhaps we can always just validate here and then rerun validation when new route versions are added? Might be missing something

Reply:
> Not sure I understand the benefit in handling this validation in the versioned router

For the versioned router this security is a function aka `RouteSecurityGetter` that gets the correct version of security on request, so we validate security config for versioned route both on `addVersion` and route registration, which we cannot do here. 
See https://github.com/elastic/kibana/pull/191973/files#diff-e700e121788da393439b09ccb5dcd68c9fe4bb6b2663f84509e1fc0cdbcfdba9R105

Does that makes sense?

Reply:
Yeah, that does make sense. I thought since the validation function takes deep partial we could do:

```ts
security: validRouteSecurity(route.security as DeepPartial<RouteSecurity>),
```

And then in `addVersion` we do it again but against the partial provided later. But maybe the security config needs all values specified together?